src/main/java/org/mockito/internal/creation/bytebuddy/CachingMockBytecodeGenerator.java

```diff
@@ -133,4 +163,60 @@ public int hashCode() {
             }
         }
     }
+
+    private interface Key<T> {
+
+        T get();
+    }
+
+    private static class StrongKey<T> implements Key<T> {
```

Comment:
I think we could force `T` to be `ClassLoader`, as this is the only usage of those key objects.
Also `StrongKey` is only used for lookup, maybe we could rename it this way.

What do you think ?